pkg/cmdexecutor/cmdexecutor.go

```diff
@@ -0,0 +1,110 @@
+package cmdexecutor
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/portworx/sched-ops/k8s"
+	"github.com/sirupsen/logrus"
+	"github.com/skyrings/skyring-common/tools/uuid"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	statusFileFormat   = "/tmp/stork-cmd-done-%s"
+	cmdWaitFormat      = "touch %s && tail -f /dev/null;"
+	cmdStatusFormat    = "stat %s"
+	killFile           = "/tmp/killme"
+	waitScriptLocation = "/tmp/wait.sh"
+	waitCmdPlaceholder = "${WAIT_CMD}"
+)
+
+const (
+	cmdStatusCheckInitialDelay = 2 * time.Second
+	cmdStatusCheckFactor       = 1
+)
+
+// StartAsyncPodCommand starts the given command in the given pod async and returns a status file
+// inside the pod that will be present if the command succeeded.
+func StartAsyncPodCommand(podNamespace, podName, container, command string) (string /*status file */, error) {
```

Comment:
Can we create a struct for this.
Executor.Init(namespace, podName, containerName, command) (error)
Executor.Start() (error) -> Starts the command and saves the filename internall
Executor.Wait() (error) -> Waits for the command to finish.

The caller of the executors shouldn't know the detail implementations.